cpp/dolfinx/common/DolfinXException.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
```

Comment:
Rename the file `exception.h` for consistency with the standard library.

Reply:
I added the suggestion